lib/dpl/provider/deis.rb

```diff
@@ -44,6 +44,31 @@ def setup_git_ssh(path, key_path)
         unless context.shell "./deis git:remote --app=#{option(:app)}"
           error 'Adding git remote failed.'
         end
+
+        wait_for_git_access
+      end
+
+      def wait_for_git_access()
+        retry_count=0
+        max_retries=30
+
+        #Parse out the git remote host and port
+        git_remote=`cat .git/config  | grep -A1 '[remote] "deis"' | tail -n 1 | sed 's/.*url = //'`
+        remote_uri=git_remote.split("ssh://")[1].split("/")[0]
```

Comment:
I fixed the test to mock out these calls so that it gets a host/port. I think in any real-world scenario the `deis git:remote --app` call that happens before this should set up the deis remote. If it doesn't then the deploy is going to bomb out somewhere either way.